### DISCUSSION TITLE
Update aio-stalwart to latest stalwart version (v0.6.0)

### DISCUSSION THREAD

**Discussion by AshanCG** [nextcloud/all-in-one#4241] [Questions] [answered]
Hi team,

After enabling stalwart with caddy (both community containers), we ran the `sudo docker exec -it nextcloud-aio-stalwart configure.sh` per https://github.com/nextcloud/all-in-one/tree/main/community-containers/stalwart#notes instructions and after that we get following error;

`Stalwart container started`
`Invalid configuration file: Invalid throttle key "remote_ip" found in "session.throttle.0000.key.0000"`.

We have tested this with isolate installation using `stalwartlabs/mail-server:v0.5.3` and we observe the same issue.

 stalwartlabs/mail-server  have issued v0.6.0 (https://github.com/stalwartlabs/mail-server/releases/tag/v0.6.0) few days ago. We tested isolated `stalwartlabs/mail-server:v0.6.0` and it works perfectly for us.
 
 At the moment, aio-stalwart set   `FROM stalwartlabs/mail-server:v0.5.3` - https://github.com/marcoambrosini/aio-stalwart/blob/main/Dockerfile. We suggest, in order to fix the above issue, to update the aio-stalwart with the latest stalwart version (v0.6.0).

**Comment by szaimen** [comment, marked as answer]
cc @marcoambrosini

**Comment by AiroPi** [comment]
Upgrading to v0.6.0 isn't enough. (At least, it wasn't for me)

Check [the upgrading guide](https://github.com/stalwartlabs/mail-server/blob/main/UPGRADING.md#upgrading-from-v053-to-v060).

Here is some step-by-step explanations on how yoy could fix the issue :
- Stop stalwart: `docker stop nextcloud-aio-stalwart`
- Create a new image: `docker commit nextcloud-aio-stalwart debug/stalwart` (probably unnecessary, but simpler)
- Enter with bash: `docker run -v nextcloud_aio_stalwart:/opt/stalwart-mail -it --rm --entrypoint bash debug/stalwart`
- Go to the config location: `cd /opt/stalwart-mail/etc`
- Remove the old `smtp` config folder (⚠️ only if you used the defaults values, otherwise, you will have to do more efforts to merge the config files correctly): `rm -rf smtp`
- Create a new smtp folder and enter it: `mkdir smtp && cd smtp`
- Download the new default config files: `curl -O "https://raw.githubusercontent.com/stalwartlabs/mail-server/main/resources/config/smtp/{auth,listener,milter,queue,remote,report,resolver,session,signature,spamfilter}.toml"`
- Exit: `exit`
- Start stalwart: `docker start nextcloud-aio-stalwart`
- Remove the previously created image: `docker image rm debug/stalwart`

Maybe this is not the most optimal solution, but it is a solution that worked for me! I hope it can help someone else...
